11_async.md

```diff
@@ -249,91 +140,47 @@ defineRequestType("note", (nest, content, source, done) => {
-a given nest. Our implementation calls `console.log` so that we can
-verify that the request arrived. Nests have a `name` property that
-holds their name.
+`defineRequestType` 函数定义了新的请求类型。以上示例为 `"note"` 请求增添了支持，该请求仅向一个给定的鸟巢发送一段笔记。我们的代码调用了 `console.log` 以便确认请求是否到达。鸟巢有一个含有它们名称的 `name` 属性。
 
 {{index "asynchronous programming"}}
 
-The fourth argument given to the handler, `done`, is a callback
-function that it must call when it is done with the request. If we had
-used the handler's ((return value)) as the response value, that would
-mean that a request handler can't itself perform asynchronous actions.
-A function doing asynchronous work typically returns before the work
-is done, having arranged for a callback to be called when it
-completes. So we need some asynchronous mechanism—in this case,
-another ((callback function))—to signal when a response is available.
-
-In a way, asynchronicity is _contagious_. Any function that calls a
-function that works asynchronously must itself be asynchronous, using
-a callback or similar mechanism to deliver its result. Calling a
-callback is somewhat more involved and error-prone than simply
-returning a value, so needing to structure large parts of your program
-that way is not great.
-
-## Promises
-
-Working with abstract concepts is often easier when those concepts can
-be represented by ((value))s. In the case of asynchronous actions, you
-could, instead of arranging for a function to be called at some point
-in the future, return an object that represents this future event.
+传递给处理器的第四个参数 `done` 是一个回调函数，处理器在处理完请求之后必须调用该函数。如果我们将处理器的((返回值))作为响应的值，那意味着一个请求处理器本身无法执行异步操作。一个做异步操作的函数通常在工作结束之前就返回，且已安排了一个在结束时就调用的回调函数。因此，我们需要一些异步的机制 —— 在这个情况下，再来一个((回调函数)) —— 以在响应就绪的时候发出信号。
+
+从某种程度上来说，异步性是 _可传染的_。对于一个以异步方式工作的函数，任何一个调用它的函数自身也必须是异步的，且使用一个回调或类似的机制来呈现其结果。与只是简单地返回一个值相比，调用一个回调要更为复杂，也更容易出错。因此，需要以这种方式构建你大部分的程序并不是一个好方法。
+
+## Promise
+
+在与抽象的概念打交道的时候，当这些概念可以被((值))所表示的话，通常会变得更为简单。在异步操作的情况里，与其说安排一个在未来某个时间点可以被调用的函数，你可以返回一个代表这种未来事件的对象。
 
 {{index "Promise class", "asynchronous programming"}}
 
-This is what the standard class `Promise` is for. A _promise_ is an
-asynchronous action that may complete at some point and produce a
-value. It is able to notify anyone who is interested when its value is
-available.
+这正是标准类 `Promise` 的用途。一个 _promise(译者注：承诺)_ 是一个可能在未来的某个时间点完成并返回一个值的异步操作。它有能力在其值可用时通知所有对此感兴趣的程序。
 
 {{index "Promise.resolve function", "resolving (a promise)"}}
 
-The easiest way to create a promise is by calling `Promise.resolve`.
-This function ensures that the value you give it is wrapped in a
-promise. If it's already a promise, it is simply returned—otherwise,
-you get a new promise that immediately finishes with your
-value as its result.
+想要创建一个 promise，最简单的办法是调用 `Promise.resolve`。这个函数会确保你给它的值被包装在一个 promise 中。如果该值已经是一个 promise，那么它会被直接返回 —— 否则，你将会得到一个立即完成的新 promise，它的结果即是你的值。
 
 ```
 let fifteen = Promise.resolve(15);
 fifteen.then(value => console.log(`Got ${value}`));
```

Comment:
if we localize "Got", then the comment below (line 166) can be translated directly

Reply:
sounds good